src/components/Table/RowState.ts

```diff
@@ -161,6 +177,30 @@ export class RowState {
       localStorage.setItem(this.persistCollapse, JSON.stringify(collapsedIds));
     }
   }
+
+  private getVisibleImmediateChildrenSelectedStates(
+    children: GridDataRow<any>[],
+    map: Map<string, SelectedState>,
+  ): SelectedState[] {
+    return children
+      .filter((row) => row.id !== "header" && this.visibleRows.has(row.id))
+      .map((row) => map.get(row.id) || this.getSelected(row.id));
+  }
+
+  // Recursively traverse through rows to determine selected state of parent rows based on children
+  private setNestedSelectedStates(row: GridDataRow<any>, map: Map<string, SelectedState>): SelectedState[] {
+    if (this.visibleRows.has(row.id)) {
+      if (!row.children) {
+        return [map.get(row.id) || this.getSelected(row.id)];
```

Comment:
This could probably be just `return [this.getSelected(row.id)]` b/c I don't think we never put leaf rows into `map`?

Reply:
Yes, you're right. I will update.